pkg/pfcpsim/session/far_builder.go

```diff
@@ -0,0 +1,79 @@
+package session
+
+import (
+	"github.com/wmnsk/go-pfcp/ie"
+)
+
+type farBuilder struct {
+	farID         uint32
+	applyAction   uint8
+	method        IEMethod
+	teid          uint32
+	downlinkIP    string
+	isDownlinkFAR bool
+}
+
+// NewFARBuilder returns a farBuilder.
+func NewFARBuilder() *farBuilder {
+	return &farBuilder{}
+}
+
+func (b *farBuilder) WithID(id uint32) *farBuilder {
+	b.farID = id
+	return b
+}
+
+func (b *farBuilder) WithMethod(method IEMethod) *farBuilder {
+	b.method = method
+	return b
+}
+
+func (b *farBuilder) WithAction(action uint8) *farBuilder {
+	b.applyAction = action
+	return b
+}
+
+func (b *farBuilder) WithTEID(teid uint32) *farBuilder {
+	b.teid = teid
+	return b
+}
+
+func (b *farBuilder) WithDownlinkIP(downlinkIP string) *farBuilder {
+	b.downlinkIP = downlinkIP
+	return b
+}
+
+func (b *farBuilder) MarkAsDownlink() *farBuilder {
+	b.isDownlinkFAR = true
+	return b
+}
+
+// BuildFAR returns by default an UplinkFAR.
+// Returns a DownlinkFAR if MarkAsDownlink was invoked.
+func (b *farBuilder) BuildFAR() *ie.IE {
+	createFunc := ie.NewCreateFAR
```

Comment:
Is it possible to also add support for "Delete FAR IE" ?

Reply:
Added but I need to further investigate if its structure is correct.